frontend/src/components/pages/your_clubs/YourClubs.tsx

```diff
@@ -1,10 +1,287 @@
+  return (
+    <div className={classes.root}>
+      <div className={classes.topUtilitiesContainer}>
+        <FormControl className={classes.formControl}>
+          <InputLabel id="number-of-displayed-clubs-label">
+            Number of Displayed Clubs
+          </InputLabel>
+          <Select
+            labelId="number-of-displayed-clubs-label"
+            id="number-of-displayed-clubs"
+            value={numClubsDisplayed ? numClubsDisplayed : 0}
+            onChange={handleNumClubsChange}
+            label="Age"
+          >
+            {textNumberRepresentation.map((item, index) => (
+              <MenuItem value={item.val}>{item.text}</MenuItem>
+            ))}
+          </Select>
+          <FormHelperText>
+            The number of clubs of which you are a member to be displayed.
+          </FormHelperText>
+        </FormControl>
+        <Button
+          className={classes.button}
+          color="primary"
+          endIcon={<AddIcon />}
+          variant="contained"
+        >
+          Create Club
+        </Button>
+      </div>
+      <div className={classes.listedClubsContainer}>
+        <DisplayListedClubs clubsToDisplay={listedClubs} />
+      </div>
+    </div>
+  );
+}
+
+interface DisplayListedClubsProps {
+  clubsToDisplay: ClubProps[],
+}
+
+/**
+ * Displays up to the number of clubs that the user has requested to to
+ * the page.
+ */
+function DisplayListedClubs(props: DisplayListedClubsProps) {
+  const classes = useStyles();
+  const clubsToDisplay = props.clubsToDisplay;
+
+  /** If clubs to display is not yet defined, or 'None' are chosen to display. */
+  if (clubsToDisplay ? /** defined */ true : /** undefined */ false)  {
+    return (
+      <div className={classes.listedClubsContainer}>
+        {clubsToDisplay.map((item, index) => (
+          <div className={classes.club} key={item.name}>
+            <Box border={1} borderColor="text.primary" borderRadius={16}>
+              <div className={classes.clubContent}>
+                <h2 className={classes.clubTitle}>{item.name}</h2>
+                <LibraryBooksRoundedIcon className={classes.clubPhoto} />
+                <LoadClubDescription description={item.description} />
+                <LoadBookInfo book={item.currentBook} />
+                <LoadContentWarnings contentWarnings={item.contentWarnings} />
+                <div className={classes.break}></div>
+                <LoadButtons />
+              </div>
+            </Box>
+          </div>
+        ))}
+      </div>
+    );
+  } else {
+    return (<div></div>);
+  }
+}
+
+interface LoadClubDescriptionProps {
+  description: string;
+};
+
+/** Displays the club's description. */
+function LoadClubDescription(props: LoadClubDescriptionProps) {
+  const classes = useStyles();
+
+  return (
+    <div>
+      <div className={classes.break}></div>
+      <p className={classes.textElement}>
+        <b>Description: </b><br/>
+        {props.description}
+      </p>
+    </div>
```

Comment:
It's up to you, but it could be more "React-like" to wrap components in fragments (<>, </>) rather than empty divs (assuming you don't need to use keys or attributes).

Reply:
Point taken! Thanks for the tip!